tonic/parameters.py

```diff
@@ -0,0 +1,158 @@
+"""
+    This is a generic Parameter class, and a ChildParameter class.
+
+    The Parameter class allows us to generate a parameter object,
+    with a certain value and a name. The name space is incremental
+    by defaul. Each Parameter object has a sympy symbol with its name.
+    
+    ChildParameter allows us to create dependent parameters 
+    with their values related to the parent Parameter value. This link
+    is given by an expression.
+
+
+    TODO:
+        1. Deal with displacement fields. 
+            These are symbolic objects generated by Geometric Entities.
+            If a displacement field is assosicated with a ChildParameter,
+            the parent Parameter should take care of the dependent disp-
+            lacement field(s).
+        2. Parameter attribute is_variable
+            I can see why we might need it, but perhaps the concept
+            should be more clear.
+"""
+
+import sympy
+
+
+class Parameter:
+    _parameters_counter = 1
+
+    def __init__(self, value, is_variable=True, name="p_"):
```

Comment:
I'd make the name default to `None`, and generate a name based on the counter then. I'd also keep a set of custom names, and throw and error if a second parameter with the same name is created.

Reply:
I believe Parameter class should not check for the same name existence, though Topology shoud.